api/types/github.go

```diff
@@ -275,11 +292,41 @@ func (c *GithubConnectorV3) MapClaims(claims GithubClaims) ([]string, []string,
 		for _, team := range teams {
 			// see if the user belongs to this team
 			if team == mapping.Team {
+				if !isEnterprise {
+					usesSSO, err := c.checkOrgSSO(mapping.Organization)
+					if err != nil {
+						return nil, nil, nil, trace.Wrap(err)
+					}
+					if usesSSO {
+						return nil, nil, nil, trace.AccessDenied(
+							"Github organization %s uses SSO, please purchase a Teleport Enterprise license if you want to authenticate with this organization",
+							mapping.Organization,
+						)
+					}
+				}
+
 				roles = append(roles, mapping.Roles...)
 			}
 		}
 	}
-	return utils.Deduplicate(roles), utils.Deduplicate(kubeGroups), utils.Deduplicate(kubeUsers)
+	return utils.Deduplicate(roles), utils.Deduplicate(kubeGroups), utils.Deduplicate(kubeUsers), nil
+}
+
+// checkOrgSSO returns true if the Github organization is a Github
+// Enterprise Cloud organization that has SSO enabled. SSO is a
+// Teleport Enterprise feature so it should not be allowed for OSS.
+func (c *GithubConnectorV3) checkOrgSSO(org string) (bool, error) {
+	// A Github organization will have a "sso" page reachable if it
+	// supports SSO. There doesn't seem to be any way to get this
+	// information from the Github REST API without being an owner
+	// of the Github organization, so check if this exists instead.
+	resp, err := http.Get(fmt.Sprintf("%s/%s/%s", githubOrgsURL, org, "sso"))
```

Comment:
1. That's true, would you recommend the org be escaped with `net/url.QueryEscape`?
3. I actually renamed a Github org I created when testing this and a 200 was returned. I think that's because Go's HTTP client follows redirects by default.

Reply:
Done